### PR TITLE
Introduce distributed RunGraphs

### PR DESCRIPTION
**Pull request by vepadulano**
Second iteration of the distributed RunGraphs function. The logic is greatly simplified now.
* There is no need (at this moment) to provide separate implementations of RunGraphs for each distributed backend. Both Spark and Dask support submitting tasks from multiple Python threads. So a common ThreadPool is cleaner and also allows to use different backends at the same time in principle
* The `GetValue` method of the proxy has been split in two functions to separate its functionalities. The `execute_graph` method is called in `RunGraphs`

### CODE REVIEW

File: bindings/experimental/distrdf/python/DistRDF/__init__.py [hunk 1]
```diff
@@ -63,6 +65,55 @@ def create_logger(level="WARNING", log_path="./DistRDF.log"):
     return logger
 
 
+def RunGraphs(proxies):
```
Comment (etejedor) [inline bindings/experimental/distrdf/python/DistRDF/__init__.py:68]: Perhaps we could think of a better place where to define `RunGraphs`? In its own module? Maybe we could have a module called e.g. `API.py` at this level where we define all the calls in `distributed`?
Can we list what functions would go there? `initialize`? Where are the `include_headers`, `include_libraries`?

Comment (etejedor) [reply bindings/experimental/distrdf/python/DistRDF/__init__.py:68]: We can just merge this and then discuss the refactoring later.

**Review by etejedor — APPROVED** [review]
Very elegant solution!

File: bindings/experimental/distrdf/python/DistRDF/__init__.py (hunk 1 above)
Comment (vepadulano) [reply bindings/experimental/distrdf/python/DistRDF/__init__.py:68]: Yes this is needed at some point. Especially for the header functionality, which is currently a bit hidden in the backend object

**Comment by vepadulano** [comment]
Test failures unrelated, merging

**Pull request merged**